### PR TITLE
Improve readme and example

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@nex3 friendly ping 😄

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Cryptographic hashing functions for Dart
+## Cryptographic hashing functions for Dart
```
Comment (nex3) [inline README.md]: It doesn't make sense to make this an h2 when it's the top-level header in the document.

File: README.md [hunk 2]
```diff
@@ -18,57 +20,42 @@ To hash a list of bytes, invoke the [`convert`][convert] method on the
 objects.
 
 ```dart
-import 'package:crypto/crypto.dart';
 import 'dart:convert'; // for the UTF8.encode method
+import 'package:crypto/crypto.dart';
```
Comment (nex3) [inline README.md]: This should be separated from the `dart:` import with a newline. See [the style guide](https://www.dartlang.org/guides/language/effective-dart/style#do-place-dart-imports-before-other-imports).

Also below.

File: README.md [hunk 3]
```diff
@@ -18,57 +20,42 @@ To hash a list of bytes, invoke the [`convert`][convert] method on the
 objects.
 
 ```dart
-import 'package:crypto/crypto.dart';
 import 'dart:convert'; // for the UTF8.encode method
```
Comment (nex3) [inline README.md]: I don't think this comment is necessary.

File: README.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
-# Cryptographic hashing functions for Dart
+## Cryptographic hashing functions for Dart
+
+[![Build Status](https://travis-ci.org/dart-lang/crypto.svg?branch=master)](https://travis-ci.org/dart-lang/crypto)
 
 A set of cryptographic hashing functions implemented in pure Dart
```
Comment (nex3) [inline README.md]: The list of hash algorithms should link to their implementations' dartdoc pages.

File: README.md [hunk 5]
```diff
@@ -18,57 +20,42 @@ To hash a list of bytes, invoke the [`convert`][convert] method on the
 objects.
 
 ```dart
-import 'package:crypto/crypto.dart';
 import 'dart:convert'; // for the UTF8.encode method
+import 'package:crypto/crypto.dart';
 
 void main() {
-  var bytes = UTF8.encode("foobar"); // data being hashed
+  var bytes = UTF8.encode('foobar'); // data being hashed
```
Comment (nex3) [inline README.md]: Changing quote types adds churn for no value.

File: README.md [hunk 6]
```diff
@@ -18,57 +20,42 @@ To hash a list of bytes, invoke the [`convert`][convert] method on the
...
-
-  print("Digest as bytes: ${digest.bytes}");
-  print("Digest as hex string: $digest");
+main() async {
+  var byteStream =
+      new Stream<String>.fromIterable(['foo', 'bar']).transform(UTF8.encoder);
+
+  var digest = await byteStream
+      .transform(sha1) // Convert to a Stream<Digest>
+      .single; // There is only one
```
Comment (nex3) [inline README.md]: We should just add `Future<Digest> Hash.convertStream(Stream<List<int>> stream)`.

File: example/example.dart [hunk 7]
```diff
@@ -7,40 +7,46 @@ import 'dart:io';
...
+final _usage = 'Usage: dart example.dart <md5|sha1|sha256> <input_filename>';
 
 Future main(List<String> args) async {
   if (args == null || args.length != 2) {
     print(_usage);
     exit(1);
   }
 
-  Hash hasher;
+  var hasher = _getHasher(args[0]);
```
Comment (nex3) [inline example/example.dart]: What's with the term "hasher"? Why not just use "hash", since that's what the class is?

Comment (kevmoo) [reply example/example.dart]: Zero clue. Just trying to minimize code churn

File: README.md (hunk 6 above)
Comment (kevmoo) [reply README.md]: Zero argument here.

File: README.md (hunk 5 above)
Comment (kevmoo) [reply README.md]: The rest of the package has standardized on single quotes. Going for consistency

File: README.md (hunk 3 above)
Comment (kevmoo) [reply README.md]: ack

File: README.md (hunk 1 above)
Comment (kevmoo) [reply README.md]: Removed header entirely. Department of redundancy department.

File: README.md (hunk 2 above)
Comment (kevmoo) [reply README.md]: ack

File: README.md (hunk 4 above)
Comment (kevmoo) [reply README.md]: ack

**Comment by kevmoo** [comment]
@nex3 – PTAL

File: README.md (hunk 6 above)
Comment (nex3) [reply README.md]: Can you add this? It shouldn't be a very big change.

Comment (nex3) [reply README.md]: Can you add this? It shouldn't be a very big change.

File: example/example.dart (hunk 7 above)
Comment (nex3) [reply example/example.dart]: Since you're making changes here, can you fix it?

File: README.md (hunk 5 above)
Comment (nex3) [reply README.md]: It's not worth the effort or churn to try to be consistent about quote types.

**Pull request closed without merging**